### PR TITLE
Adding Indexing

### PR DESCRIPTION
**Pull request by jtaleric**
Store results into OpenSearch

### CODE REVIEW

File: netperf/archive.go [hunk 1]
```diff
@@ -1,36 +1,119 @@
...
+	}
+	logging.Infof("Connected to : %s\n", config.Addresses)
+	return client, nil
+}
+
+// BuildDocs returns the documents that need to be indexed or an error.
+func BuildDocs(sr ScenarioResults) ([]Doc, error) {
+	u := uuid.New()
+	uuid := fmt.Sprintf("%s", u.String())
+	time := time.Now()
```
Comment (rsevilla87) [inline netperf/archive.go:65]: Can we use UTC time?

File: netperf/archive.go [hunk 2]
```diff
@@ -1,36 +1,119 @@
...
+	Profile       string           `json:"profile"`
+	Duration      int              `json:"duration"`
+	Samples       int              `json:"samples"`
+	Messagesize   int              `json:"messageSize"`
+	Result        float64          `json:"result"`
+	Metric        string           `json:"metric"`
+	ServerNodeCPU metrics.NodeCPU  `json:"serverCPU"`
+	ServerPodCPU  []metrics.PodCPU `json:"serverPods"`
+	ClientNodeCPU metrics.NodeCPU  `json:"ClientCPU"`
+	ClientPodCPU  []metrics.PodCPU `json:"ClientPods"`
```
Comment (rsevilla87) [inline netperf/archive.go:39]: Just to follow the camelCase semanthics maybe we can rename these JSON fields?
```suggestion
	ClientNodeCPU metrics.NodeCPU  `json:"clientCPU"`
	ClientPodCPU  []metrics.PodCPU `json:"clientPods"`
```

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
just a couple of minor nits and LGTM

**Comment by jtaleric** [comment]
Thanks @rsevilla87 comments addressed!

File: netperf/archive.go [hunk 3]
```diff
@@ -1,36 +1,119 @@
...
+		d.ClientPodCPU = r.ClientPodCPU.Results
+		docs = append(docs, d)
 	}
-	return ec, nil
+	return docs, nil
+}
+
+// IndexDocs indexes results from k8s-netperf returns failures if any happen.
+func IndexDocs(client *opensearch.Client, docs []Doc) error {
+	logging.Infof("Attempting to index %d documents", len(docs))
```
Comment (mohit-sheth) [inline netperf/archive.go:98]: thoughts on adding 1 retry if it fails?

File: netperf/archive.go [hunk 4]
```diff
@@ -1,36 +1,119 @@
...
-	fmt.Printf("Connecting to ES - %s\r\n", es.url)
-	esc := elasticsearch.Config{
-		Username:  es.user,
-		Password:  es.password,
-		Addresses: []string{es.url},
+// Connect returns a client connected to the desired cluster.
+func Connect(url string, skip bool) (*opensearch.Client, error) {
+	if len(url) < 1 {
+		return nil, fmt.Errorf("URL cannot be empty")
+	}
```
Comment (mohit-sheth) [inline netperf/archive.go:46]: can remove this as we are already checking this condition above

**Review by mohit-sheth — APPROVED** [review]
LGTM

File: netperf/archive.go (hunk 3 above)
Comment (jtaleric) [reply netperf/archive.go:98]: `The Go client retries requests for a maximum of three times by default.`

Comment (jtaleric) [reply netperf/archive.go:98]: From the OpenSearch docs.

File: netperf/archive.go (hunk 4 above)
Comment (jtaleric) [reply netperf/archive.go:46]: Sure, we can depend on other calls to check the URL before sending.

**Pull request merged**